mypy/checkexpr.py

```diff
@@ -1335,6 +1345,46 @@ def check_callable_call(
             callee = callee.copy_modified(ret_type=new_ret_type)
         return callee.ret_type, callee
 
+    def _check_non_ret_from_prot(self, type: TypeInfo, attr_name: str) -> bool:
+        """Determine whether the attribute is a function with None-compatible return type.
+
+        Additionally, check whether the function was defined in a protocol.
+        """
+        if not state.strict_optional:
+            # If strict-optional is not set, is_subtype(NoneType(), T) is always True.
+            # So, we cannot do anything useful here in that case.
+            return False
+        for base in type.mro:
+            symnode = base.names.get(attr_name)
+            if symnode is None or not base.is_protocol:
+                continue
+            node = symnode.node
+            if isinstance(node, OverloadedFuncDef):
+                if node.impl is not None:
+                    typ = get_proper_type(node.impl.type)
+                    assert isinstance(typ, CallableType)
+                    return is_subtype(NoneType(), typ.ret_type)
+                for func in node.items:
+                    if isinstance(func, Decorator):
+                        func = func.func
+                    assert isinstance(func.type, CallableType)
+                    if is_subtype(NoneType(), func.type.ret_type):
+                        return True
+                    if func.is_property:
+                        return False  # Only check the first overload of properties.
+                else:
+                    return False
+            if isinstance(node, Decorator):
+                node = node.func
+            if isinstance(node, FuncDef):
+                if node.type is not None:
+                    assert isinstance(node.type, CallableType)
+                    return is_subtype(NoneType(), node.type.ret_type)
+                else:
+                    # No type annotations are available
+                    return True
```

Comment:
I think we should return False here, don't add the note if we are not sure what is the intended return type. You can simply delete the `else` branch so we fall through to last `return False`.